src/ray/object_manager/pull_manager.cc

```diff
@@ -131,7 +135,12 @@ void PullManager::Tick() {
     const auto time = get_time_();
     if (time >= request.next_pull_time) {
       TryPull(object_id);
-      request.next_pull_time = time + pull_timeout_ms_ / 1000;
+      request.num_retries++;
+      // Better to fail than have unexpected overflow behavior.
+      RAY_CHECK(false) << "Pull manager has retried pull object: " << object_id
```

Comment:
Hmm a bit confused about this message, won't this just trigger immediately?

Also, I think it's better to just cap at a maximum timeout of say ~60s and not rely on the assertion. Otherwise, the pulls will just take so long that it will basically be like the system has come to a standstill.

Reply:
> Hmm a bit confused about this message, won't this just trigger immediately?

Whoops sorry, that was just a careless last second edit. 

> Also, I think it's better to just cap at a maximum timeout of say ~60s and not rely on the assertion. 

This isn't really an exponential timeout then right? It would just go 10, 20. 40, 60, 60,...